heron/spi/src/java/com/twitter/heron/spi/statemgr/SchedulerStateManagerAdaptor.java

```diff
@@ -122,7 +122,9 @@ public Boolean setSchedulerLocation(
    * @return Boolean - Success or Failure
    */
   public Boolean deleteTMasterLocation(String topologyName) {
-    return awaitResult(delegate.deleteTMasterLocation(topologyName));
+    return awaitResult(delegate.tmasterLocationExists(topologyName))
+        ? awaitResult(delegate.deleteTMasterLocation(topologyName))
+        : true;
```

Comment:
The other way I could do this is surround these all with try/catch blocks and specifically catch zookeeper exceptions. I dont know if there is a "safe" version of the zookeeper interaction where if the node does not exist it doesnt throw an exception but I can look.